### PR TITLE
README: Re-introduce a NEMU specific README

### LINKED ISSUE #69: update README
Is this a fork of Qemu?  Why?  Should have a unique description in the README.

### PR DESCRIPTION
**Pull request by sameo**
This README markdown file is based on the topic/compile-reduced initial
NEMU README.

Based on the evolution of the project since the early NEMU days, this
new README version comes with:

- A small description of the virt i386 machine type.
- An updated list of guest platforms
- An updated list of guest OSes
- A shortened list of boot methods and firmware type support
- A slightly updated device model description

Fixes #69

Signed-off-by: Samuel Ortiz <sameo@linux.intel.com>

### CODE REVIEW

**Comment by sameo** [comment]
Diff between this README.md and the topic/compile-recduced one:

```diff
diff --git a/README.md b/README.md
index fa42dda749..6835d33722 100644
--- a/README.md
+++ b/README.md
@@ -1,3 +1,25 @@
+- [NEMU, a cloud hypervisor](#nemu--a-cloud-hypervisor)
+  * [Rationale](#rationale)
+  * [QEMU base](#qemu-base)
+  * [*NEMU*](#-nemu-)
+  * [Requirements](#requirements)
+    + [High Level](#high-level)
+    + [Architectures](#architectures)
+    + [Guest OS](#guest-os)
+    + [Guest Platforms](#guest-platforms)
+    + [Host Platforms](#host-platforms)
+    + [Firmware and boot](#firmware-and-boot)
+    + [Boot methods](#boot-methods)
+    + [Memory](#memory)
+    + [Devices](#devices)
+      - [Models](#models)
+      - [Block](#block)
+    + [Guest Image Formats](#guest-image-formats)
+    + [Migration](#migration)
+    + [Monitoring](#monitoring)
+    + [**To be discussed**](#--to-be-discussed--)
+
+
 # NEMU, a cloud hypervisor
 
 *NEMU* is an open source hypervisor specifically built and designed to run modern cloud
@@ -30,8 +52,8 @@ allows for easier review, fuzz testing, modularization and future innovation.
 
 QEMU is the current de facto standard open source cloud hypervisor. It has a rich set of features
 that have been developed and tested over time. This includes features such as live migration,
-PCI, Memory and CPU hotplug, VFIO, mediated device passthrough and vhost-user. QEMU also has been
-the code base on which significant effort and innovation has been invested to create multiple
+PCI, Memory, NVDIMM and CPU hotplug, VFIO, mediated device passthrough and vhost-user. QEMU also
+has been the code base on which significant effort and innovation has been invested to create multiple
 performant I/O models
 
 It also comes with a very large support for legacy features, for platforms and devices and is capable
@@ -57,23 +79,28 @@ of generic abstractions.
 
 ## Requirements
 
-*NEMU* provides a PCI virt-io platform with support for vfio based device direct assigment and mediated
-device assigment support. It also aims to retain support for live migration, device, memory and CPU
-hotplug and vhost-user. *NEMU* will need to emulate a small subset of feature including pci host brige,
-pci-pci bridges, cdrom and PAM.
+*NEMU* provides a PCI virtio platform with support for vfio based device direct assigment and mediated
+device assigment support. It also aims to retain support for live migration, PCI device, memory, NVDIMM and
+CPU hotplug and vhost-user. *NEMU* will need to emulate a small subset of feature including PCI host brige.
+
+*NEMU* also introduces a new QEMU x86-64 machine type: **`virt`**. It is a purely virtual platform, that
+does not try to emulate any existing x86 chipset or legacy bus (ISA, SMBUS, etc) and offloads as many features
+to KVM as possible. This is a similar approach as the already existin AArch64 `virt` machine type and *NEMU*
+will only support the two `virt` machine types.
 
-Below is a list of QEMU features that *NEMU* will retain.
+Below is a list of QEMU features that *NEMU* will retain and add.
 
 ### High Level
 
 * KVM and KVM only based
-* No emulation
+* Minimal emulation
 * Low latency
 * Low memory footprint
 * Low complexity
 * Small attack surface
 * 64-bit support only
 * Direct Boot support
+* CPU, memory, PCI and NVDIMM hotplug
 * Machine to machine migration
 
 ### Architectures
@@ -82,16 +109,13 @@ Below is a list of QEMU features that *NEMU* will retain.
 * `x86-64`
 * `AArch64`
 
-**TODO** Define an exhaustive set of supported architectures (ARMv8.x, *lake)
-
 ### Guest OS
 * `64-bit Linux`
-* `64-bit Windows Server`
+* `64-bit Windows Server (headless, Nano Server)`
 
 ### Guest Platforms
 
-* `Q35` (x86-64) Pure PCI-E platform
-* `PIIX4` (x86-64) With ACPI support
+* `virt` (x86-64) QEMU x86-64 virtual machine
 * `virt` (AArch64) QEMU AArch64 virtual machine
 
 ### Host Platforms
@@ -101,12 +125,12 @@ Below is a list of QEMU features that *NEMU* will retain.
 ### Firmware and boot
 
 * `UEFI`
-* `qboot`
 * `ACPI`
-  * ACPI tables generation
+  * Hardware Reduced ACPI
   * Hotplug support
        * CPU
        * Memory
+       * NVDIMM
        * PCI devices
        * VFIO
        * vhost-user
@@ -114,10 +138,6 @@ Below is a list of QEMU features that *NEMU* will retain.
 ### Boot methods
 
 * `UEFI boot`
-* `qboot boot`
-* `Direct kernel boot`
-  * bzImage
-  * ELF Kernel
 
 ### Memory
 
@@ -139,6 +159,7 @@ Below is a list of QEMU features that *NEMU* will retain.
   * `scsi`
        * `virtio`
        * `vhost`
+  * `9pfs`
   * `vhost-user-scsi`
   * `vhost-user-net`
   * `vhost-user-blk`
@@ -152,16 +173,20 @@ Below is a list of QEMU features that *NEMU* will retain.
 * `TPM`
   * `vTPM`
   * `Host TPM passthrough`
+* `SCSI controller`
+* `PCI controller (pci-lite)`
 
 #### Block
 
 * `cdrom`
 * `nvme`
+* `ceph/rbd`
 
 ### Guest Image Formats
 
 * `QCOW2`
 * `RAW`
+* `VHD`
 
 ### Migration
 
@@ -173,16 +198,12 @@ Below is a list of QEMU features that *NEMU* will retain.
 * `QMP`
 * `QAPI`
 
-### Others
-
 ### **To be discussed**
 
+* `qboot`
 * `Graphic Console`
 * `virtio-block-crypto`
-* `SCSI controller`
-* `SATA controller`
 * `QEMU client support as modules`
-  * `rbd`
   * `iscsi`
   * `nbd`
   * `nfs`
@@ -190,4 +211,3 @@ Below is a list of QEMU features that *NEMU* will retain.
 * `RDMA live migration`
 * `SLIRP`
 * `Guest agent`
-* `virtio-9pfs`
```

File: README.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+### High Level
+
+* KVM and KVM only based
+* Minimal emulation
+* Low latency
+* Low memory footprint
+* Low complexity
+* Small attack surface
+* 64-bit support only
+* Direct Boot support
```
Comment (liujing2) [inline README.md]: Do we still support direct boot, meaning nofw?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+The goal of *NEMU* is to retain the absolute minimal subset of the QEMU codebase that is required
+for the feature set described below. The QEMU code base will also be simplified to reduce the number
+of generic abstractions.
+
+## Requirements
+
+*NEMU* provides a PCI virtio platform with support for vfio based device direct assigment and mediated
+device assigment support. It also aims to retain support for live migration, PCI device, memory, NVDIMM and
+CPU hotplug and vhost-user. *NEMU* will need to emulate a small subset of feature including PCI host brige.
```
Comment (liujing2) [inline README.md]: If my poor English makes a mistake in comment, please ignore it :)
a small subset of s/feature/features

File: README.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+### Memory
+
+* `QEMU allocated memory`
+* `File mapped memory`
+* `Huge pages`
+* `Memory pinning`
+
+### Devices
+
```
Comment (liujing2) [inline README.md:149]: We have virtio-serial-pci and virtconsole devices? And pci-lite host bridge? I am not sure what is the difference between Devices and Models.

**Comment by sameo** [comment]
@liujing2 PR updated, thanks.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+### Architectures
+
+*NEMU* only supports two 64-bit CPU architectures:
+* `x86-64`
+* `AArch64`
+
+### Guest OS
+* `64-bit Linux`
+* `64-bit Windows Server (headless, Nano Server)`
```
Comment (rbradford) [inline README.md]: Nano Server is only available as a container image now.

**Comment by sameo** [comment]
@rbradford PR updated, I moved the windows server support to the `to be discussed` section.

File: README.md [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+### Host Platforms
+
+* `Linux`
+
+### Firmware and boot
+
+* `UEFI`
+* `ACPI`
+  * Hardware Reduced ACPI
+  * Hotplug support
```
Comment (mcastelino) [inline README.md]: We should keep the hot plug requirements separate and distinct from the minimal requirement for NEMU. The smallest simplest platform type does not require hotplug. So we need to be able to support (potentially at build time) a minimal platform that does not support hotplug (and maybe even drop NVDIMM in that platform which always requires hotplug), but has the lowest complexity and code footprint.

**Review by mcastelino — APPROVED** [review]
@sameo as I mentioned earlier the hotplug support should be separated from the minimal requirements of QEMU, and compiling out support for hotplug should be provided as a build option. Otherwise LGTM.

**Comment by mcastelino** [comment]
LGTM

**Pull request merged**